eitprocessing/features/breath_detection.py

```diff
@@ -0,0 +1,425 @@
+                continue
+
+            current_valley_index += 1
+
+        return peak_indices, valley_indices
+
+    def _remove_low_amplitudes(
+        self,
+        data: np.ndarray,
+        peak_indices: np.ndarray,
+        valley_indices: np.ndarray,
+    ) -> tuple[np.ndarray, np.ndarray]:
+        """
+        Remove peaks if the amplitude is low compared to the median amplitude.
+
+        The amplitude of a peak is determined as the average vertical distance
+        between the peak value and the two valley values besides it. The cutoff
+        value for the amplitude is calculated as the median amplitude times
+        `amplitude_cutoff_fraction`. Peaks that have an amplitude below the
+        cutoff are removed. Then, `_remove_doubles()` is called to remove
+        either of the valleys next to the peak.
+
+        If `amplitude_cutoff_fraction` is None, the input is returned
+        unchanged.
+
+        Args:
+            data: the data the peaks and valleys were found in
+            peak_indices (np.ndarray): the indices of the peaks
+            valley_indices (np.ndarray): the indices of the valleys
+
+        Returns:
+            A tuple (peak_indices, peak_values, valley_indices, valley_values)
+            with low-amplitude breaths removed.
+        """
+        if not self.amplitude_cutoff_fraction:
+            return peak_indices, valley_indices
+
+        peak_values = data[peak_indices]
+        valley_values = data[valley_indices]
+
+        inspiratory_amplitude = peak_values - valley_values[:-1]
+        expiratory_amplitude = peak_values - valley_values[1:]
+        amplitude = (inspiratory_amplitude + expiratory_amplitude) / 2
+
+        amplitude_cutoff = self.amplitude_cutoff_fraction * np.median(amplitude)
+        delete_peaks = np.argwhere(amplitude < amplitude_cutoff)
+
+        peak_indices = np.delete(peak_indices, delete_peaks)
+        peak_values = np.delete(peak_values, delete_peaks)
+
+        return self._remove_doubles(data, peak_indices, valley_indices)
+
+    def _remove_breaths_around_invalid_data(
+        self,
+        breaths: list[Breath],
+        time: np.ndarray,
+        outliers: np.ndarray,
+    ) -> list[Breath]:
+        """Remove breaths overlapping with invalid data.
+
+        This function defines a lower and upper cutoff. Data beyond those cutoffs is considered invalid for the
+        purposes of breath detection. Breaths that start within a window length of invalid data are removed.
+
+        The lower cutoff is a distance away from the mean. The distance is m times the distance between the
+        mean and the nth percentile of the data. m is given by `invalid_data_removal_multiplier` and n is given by
+        `invalid_data_removal_percentile`.
+
+        For example, with m = 4 and n = 5, the mean = 100, and 5% of the data is below/equal to 90, all data below 100 -
+        (4 * 10) = 60 and above 100 + (4 * 10) = 140 is considerd invalid.
```

Comment:
No. n indicates the nth percentile, i.e. 5% and 95%. However, there was another issue. The lower and upper limit both depend on different extremes. I adapted the example accordingly.

 The docstring was also outdated, because I split this into multiple functions. I moved some documentation around.

5dda8daaa31378c8f8567d34ceb7ac717d0396f9

Reply:
I also refactored the names of variables slightly in 07e9d7919fb773474f6e2a99a855d18cc3fd68fa